sn_logging/src/lib.rs

```diff
@@ -17,18 +17,66 @@ use layers::TracingLayers;
 use std::path::PathBuf;
 use tracing::info;
 use tracing_appender::non_blocking::WorkerGuard;
-use tracing_core::{dispatcher::DefaultGuard, Level};
+use tracing_core::dispatcher::DefaultGuard;
 use tracing_subscriber::{prelude::__tracing_subscriber_SubscriberExt, util::SubscriberInitExt};
 
 pub use error::Error;
 pub use layers::ReloadHandle;
 
+// re-exporting the tracing crate's Level as it is used in our public API
+pub use tracing_core::Level;
+
 #[derive(Debug, Clone)]
 pub enum LogOutputDest {
     Stdout,
     Path(PathBuf),
 }
 
+fn curent_exe_name() -> String {
+    std::env::args()
+        .next()
+        .and_then(|arg| {
+            std::path::Path::new(&arg).file_name().map(|s| {
+                let mut name = s.to_string_lossy().into_owned();
+                if cfg!(windows) && name.to_lowercase().ends_with(".exe") {
+                    name.truncate(name.len() - 4); // Remove the last 4 characters (.exe)
+                }
+                name
+            })
+        })
+        .unwrap_or_else(|| "default".to_string())
+}
+
+impl LogOutputDest {
+    pub fn parse_from_str(val: &str) -> Result<Self> {
+        match val {
+            "stdout" => Ok(LogOutputDest::Stdout),
+            "data-dir" => {
+                // Get the current timestamp and format it to be human readable
+                let timestamp = chrono::Local::now().format("%Y-%m-%d_%H-%M-%S").to_string();
+
+                // Get the data directory path and append the timestamp to the log file name
+                let dir = match dirs_next::data_dir() {
+                    Some(dir) => dir
+                        .join("safe")
+                        .join(curent_exe_name())
+                        .join("logs")
+                        .join(format!("log_{timestamp}")),
```

Comment:
Log to `safe/[exe_name]/logs/`
so any app can use it and the path depends on app name.